src/ast/values.rs

```diff
@@ -267,6 +287,62 @@ impl<'a> From<i32> for ParameterizedValue<'a> {
     }
 }
 
+impl<'a> TryFrom<ParameterizedValue<'a>> for i64 {
+    type Error = &'static str;
+
+    fn try_from(value: ParameterizedValue<'a>) -> Result<i64, Self::Error> {
+        match value {
+            ParameterizedValue::Integer(v) => Ok(v),
+            _ => Err("Not an integer")
+        }
+     }
+}
+
+impl<'a> TryFrom<ParameterizedValue<'a>> for f64 {
+    type Error = &'static str;
+
+    fn try_from(value: ParameterizedValue<'a>) -> Result<f64, Self::Error> {
+        match value {
+            ParameterizedValue::Real(v) => Ok(v),
```

Comment:
Maybe use the dedicated conversion methods from `ParameterizedValue` here and return an exact `Error` from this crate's error type.